### ISSUE TITLE
TLS activation on latest VM 21.0.1 fails with exception in certbot HTTP challenge

### ISSUE THREAD

**Issue by tig987** [nextcloud/vm#1943]
## Steps to reproduce
1. dowload latest VM image 21.0.1
2. run first installation
3. Activate TLS

## Actual behavior
certbot HTTP challenge fails with exception

## Server configuration

VM ova image 21.0.1

**Nextcloud server version:** (see your admin page)

**Server OS** (Ubuntu server is default)

**PHP Version** (e.g. 7.4)

**How did you install the VM?** (Scripted install from master **OR** Released version)

## Network
**Do you use DHCP?**
yes
**Is port 80 and/or 443 open?**
yes
## Logs / Screenshots

```
DNS seems correct when checking with nslookup!
DNS seems correct when checking with dig!
Testing if network is OK...
Checking connection...
Checking if port 80 is open with https://www.networkappers.com/tools/open-port-checker...
Port 80 is open on 123.456.789.123!
Checking if port 443 is open with https://www.networkappers.com/tools/open-port-checker...
certbot 1.15.0
certbot 1.15.0
Getting current PHP-version...
PHPVER=7.4
/etc/apache2/sites-available/nc.dummy.domain.conf was successfully created.
1085
1313
1315
1325
Site 000-default already disabled
apache2.service is not active, cannot reload.
Trying to generate certs and validate them with standalone method.
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator standalone, Installer None
Running pre-hook command: systemctl stop apache2.service
Requesting a certificate for nc.dummy.domain
Performing the following challenges:
http-01 challenge for nc.dummy.domain
Waiting for verification...
----------------------------------------
Exception happened during processing of request from ('::ffff:10.87.35.1', 52486, 0, 0)
Traceback (most recent call last):
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/snap/certbot/1150/lib/python3.8/site-packages/acme/standalone.py", line 207, in __init__
    BaseHTTPServer.BaseHTTPRequestHandler.__init__(self, *args, **kwargs)
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 720, in __init__
    self.handle()
  File "/snap/certbot/1150/lib/python3.8/site-packages/acme/standalone.py", line 232, in handle
    BaseHTTPServer.BaseHTTPRequestHandler.handle(self)
  File "/snap/certbot/1150/usr/lib/python3.8/http/server.py", line 427, in handle
    self.handle_one_request()
  File "/snap/certbot/1150/usr/lib/python3.8/http/server.py", line 395, in handle_one_request
    self.raw_requestline = self.rfile.readline(65537)
  File "/snap/certbot/1150/usr/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
ConnectionResetError: [Errno 104] Connection reset by peer
----------------------------------------
----------------------------------------
Exception happened during processing of request from ('::ffff:10.87.35.1', 34514, 0, 0)
Traceback (most recent call last):
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 316, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 347, in process_request
    self.finish_request(request, client_address)
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 360, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/snap/certbot/1150/lib/python3.8/site-packages/acme/standalone.py", line 207, in __init__
    BaseHTTPServer.BaseHTTPRequestHandler.__init__(self, *args, **kwargs)
  File "/snap/certbot/1150/usr/lib/python3.8/socketserver.py", line 720, in __init__
    self.handle()
  File "/snap/certbot/1150/lib/python3.8/site-packages/acme/standalone.py", line 232, in handle
    BaseHTTPServer.BaseHTTPRequestHandler.handle(self)
  File "/snap/certbot/1150/usr/lib/python3.8/http/server.py", line 427, in handle
    self.handle_one_request()
  File "/snap/certbot/1150/usr/lib/python3.8/http/server.py", line 395, in handle_one_request
    self.raw_requestline = self.rfile.readline(65537)
  File "/snap/certbot/1150/usr/lib/python3.8/socket.py", line 669, in readinto
    return self._sock.recv_into(b)
ConnectionResetError: [Errno 104] Connection reset by peer
----------------------------------------
Challenge failed for domain nc.dummy.domain
http-01 challenge for nc.dummy.domain
Cleaning up challenges
Running post-hook command: systemctl start apache2.service
post-hook command "systemctl start apache2.service" returned error code 1
Error output from post-hook command systemctl:
Job for apache2.service failed because the control process exited with error code.
See "systemctl status apache2.service" and "journalctl -xe" for details.

Some challenges have failed.
```

#### Log file (/var/log/nextcloud/nextcloud.log)
```
Insert your nextcloud.log file here
```

#### Installation log (command output)
```
Insert the CLI output here
```

**Comment by szaimen** [comment]
@enoch85 this looks like an issue with IPv6... 🤔

**Comment by int-red** [comment]
just for your information.
I do not use IPv6 in my network. My static WAN IP as well as my local LAN are IPv4.

**Comment by enoch85** [comment]
I just ran the script with a customer, and no errors. 

So I guess you need to take a look at your network configuration @int-red You can find more info here: https://docs.hanssonit.se/s/W6fMouPiqQz3_Mog/virtual-machines-vm/d/W6-83ePiqQz3_MrT/publish-your-server-online

If that doesn't work, please try with deSEC instead.

**Comment by UNlDAN** [comment]
Can you please make sure TCP AND UDP are being forwarded to ports 80 and 443? Sounds weird but I have dealt with this issue before.....TCP only on 443 did not work.

**Comment by tig987** [comment]
I was able to fix the issue.
I am using HAproxy as reverse proxy on my firewall to forward the letsencrypt HTTP challenges to my nextcloud server.
The HAProxy Backend for my nextcloud server had the Health check method "basic" active which marked the service at port 80 as not available. Therefore the request from letsencrypt always got a 503 service not available from HAproxy.
I disabled the Health check method so that port 80 would be always be forwarded.
This solved the issue and certbot is working now.
